File: website/Entities/Student.cs
using System;

namespace website.Entities
{
    public class Student : IAuditFields
    {
        public string BarCode { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public DateTime CreatedDate { get; set; }
        public DateTime ModifiedDate { get; set; }
    }
}

File: HomeReadingLibraryWeb/ClientApp/src/app/entities/student.ts
export class Student {
    firstName: string;
    lastName: string;
    barCode?: string;

    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    public static fromObject(studentObj: any): Student {
        const student = new Student();
        Object.assign(student, studentObj);

        return student;
    }
}
